Modules/_decimal/_decimal.c

```diff
@@ -5806,19 +5833,22 @@ PyInit__decimal(void)
 
 
     /* Init types */
-    PyDec_Type.tp_base = &PyBaseObject_Type;
-    PyDecContext_Type.tp_base = &PyBaseObject_Type;
-    PyDecContextManager_Type.tp_base = &PyBaseObject_Type;
-    PyDecSignalDictMixin_Type.tp_base = &PyBaseObject_Type;
+#define CREATE_TYPE(mod, type, spec)                                        \
+    do {                                                                    \
+        type = (PyTypeObject *)PyType_FromMetaclass(NULL, mod, spec, NULL); \
+        CHECK_PTR(type);                                                    \
+    } while (0)
```

Comment:
Nitpick; I prefer the condensed variant:

```suggestion
#define CREATE_TYPE(MOD, TP, SPEC) do {                               \
    TP = (PyTypeObject *)PyType_FromMetaclass(NULL, MOD, SPEC, NULL); \
    CHECK_PTR(TP);                                                    \
} while (0)
```

Reply:
Looks good!

But macro parameters are usually named in the same style as function parameters, e.g. unix like style and camelCase, IMO. This helps us distinguish macro parameters from other macros or constants such as `NULL`.

Do you mind if I modify the style of macro parameters. :)

Reply:
Well, it's a personal preference ;) You'll find both styles in the CPython code base. I've come to like the upper case spelling for the following reasons:

- I find it easier to read (personal preference)
- Most variable names are lower case; using upper case inside the macro prevents you from accidentally shadowing variables in your surrounding scope

I said my preference; do as you wish ;)